PrestaShop/PrestaShop
# Deprecate Domain\SpecificPrice namespace

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | See below
| Type?             | refacto
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | yes
| Fixed ticket?     | 
| How to test?      | See below
| Possible impacts? | Order creation back office page logic for modifying product price in cart

### Description

Namespace `Domain\Product\SpecificPrice\*` was introduced during product page migration, but not everything was deprecated in namespaces used before this one.

This Pull Request deprecates everything in `Domain\SpecificPrice\*` and references to a `\Domain\Product\SpecificPrice\*` .

It also replaces following commands usage in `cartController:edtiProductPriceAction()` : `AddSpecificPriceCommand` & `DeleteSpecificPriceByCartProductCommand` are replaced by a single command specially made for that usecase - `UpdateProductPriceInCartcommand` which does the same under the hood.

### How to test

On order creation back office page (`Sell -> orders -> Add new order`): make sure that _modifying product price in cart_ behavior works as previously. To be able to modify a product price in cart: search for any customer and add any product, then try to change the price of product and see if it works as expected.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/24658)
